lib/gltfPipeline.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var addDefaults = require('./addDefaults');
+var removeUnusedImages = require('../').removeUnusedImages;
+var removeUnusedSamplers = require('../').removeUnusedSamplers;
+var removeUnusedShaders = require('../').removeUnusedShaders;
+var removeUnusedTechniques = require('../').removeUnusedTechniques;
+var removeUnusedPrograms = require('../').removeUnusedPrograms;
+var removeUnusedBuffers = require('../').removeUnusedBuffers;
+var removeUnusedBufferViews = require('../').removeUnusedBufferViews;
+var removeUnusedMaterials = require('../').removeUnusedMaterials;
+var removeUnusedSkins = require('../').removeUnusedSkins;
+var removeUnusedCameras = require('../').removeUnusedCameras;
+var removeUnusedTextures = require('../').removeUnusedTextures;
+var removeUnusedMeshes = require('../').removeUnusedMeshes;
+var removeUnusedNodes = require('../').removeUnusedNodes;
+var removeUnusedAccessors = require('../').removeUnusedAccessors;
+var removeUnused = require('./removeUnused');
+var loadGltfUris = require('./loadGltfUris');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var DeveloperError = Cesium.DeveloperError;
+
+
+module.exports = gltfPipeline;
+
+function gltfPipeline(options, callback) {
```

Comment:
Since this will be the starting point for projects that use gltf-pipeline as a library, we should also accept gltf objects directly. Definitely the obj2gltf will use it this way instead of writing a file first, then calling this. Maybe there should be another file that reads gltf files? @pjcozzi what do you think?